### PR TITLE
Add User service implementation

### PR DESCRIPTION
**Pull request by SkillfulDev**
* Add All user service logic

### CODE REVIEW

File: src/main/java/online/carsharing/controller/UserController.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class UserController {
                     + " Requires the new role information to be provided in the request body."
     )
     @PutMapping("/{id}/role")
-    @PreAuthorize("hasAnyRole('ROLE_MANAGER')")
+    @PreAuthorize("hasRole('ROLE_MANAGER')")
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/controller/UserController.java]:
```suggestion
    @PreAuthorize("hasRole('MANAGER')")
```

File: src/main/java/online/carsharing/controller/UserController.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public UserResponseDto getUserProfile(@AuthenticationPrincipal User currentUser)
     @PatchMapping("/me")
     @PreAuthorize("hasAnyRole('ROLE_MANAGER','ROLE_CUSTOMER')")
```
Comment (HAIOVYI) [inline src/main/java/online/carsharing/controller/UserController.java]:
```suggestion
    @PreAuthorize("hasAnyRole('MANAGER', 'CUSTOMER')")
```

**Review by HAIOVYI — DISMISSED** [review]
GJ! Approve

**Review by nklimovych — APPROVED** [review]
Amazing!

**Review by HAIOVYI — APPROVED** [review]
best!

**Pull request merged**